File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy= ListNode()
        curr = dummy
        curr.next = head
        while head and head.next:
            tempHead1 = head
            tempHead2 = tempHead1.next
            tempHead3 = tempHead2.next
            
            curr.next = tempHead2
            tempHead2.next = tempHead1
            tempHead1.next = tempHead3
            
            curr = tempHead1
            head = tempHead3
        return dummy.next
            
            
            
        

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
       if(!(head && head->next)){
           return head;
       }
        ListNode* dummy = new ListNode();
        ListNode* curr = dummy;
        while(head && head->next){
            ListNode* temp = head->next->next;
            curr->next = head->next;
            curr = curr->next;
            head->next->next = head;
            curr = curr->next;
            curr->next = nullptr;
            head = temp;
        }
        if(head){
            curr->next = head;
        }
        return dummy->next;
    }
};